src/java.base/share/classes/jdk/internal/foreign/PlatformLayouts.java

```diff
@@ -233,42 +215,44 @@ private RISCV64() {
         /**
          * The {@code short} native type.
          */
-        public static final ValueLayout.OfShort C_SHORT = ValueLayout.JAVA_SHORT.withBitAlignment(16);
+        public static final ValueLayout.OfShort C_SHORT = ValueLayout.JAVA_SHORT;
 
         /**
          * The {@code int} native type.
          */
-        public static final ValueLayout.OfInt C_INT = ValueLayout.JAVA_INT.withBitAlignment(32);
+        public static final ValueLayout.OfInt C_INT = ValueLayout.JAVA_INT;
 
         /**
          * The {@code long} native type.
          */
-        public static final ValueLayout.OfLong C_LONG = ValueLayout.JAVA_LONG.withBitAlignment(64);
+        public static final ValueLayout.OfLong C_LONG = ValueLayout.JAVA_LONG;
 
         /**
          * The {@code long long} native type.
          */
-        public static final ValueLayout.OfLong C_LONG_LONG = ValueLayout.JAVA_LONG.withBitAlignment(64);
+        public static final ValueLayout.OfLong C_LONG_LONG = ValueLayout.JAVA_LONG;
 
         /**
          * The {@code float} native type.
          */
-        public static final ValueLayout.OfFloat C_FLOAT = ValueLayout.JAVA_FLOAT.withBitAlignment(32);
+        public static final ValueLayout.OfFloat C_FLOAT = ValueLayout.JAVA_FLOAT;
 
         /**
          * The {@code double} native type.
          */
-        public static final ValueLayout.OfDouble C_DOUBLE = ValueLayout.JAVA_DOUBLE.withBitAlignment(64);
+        public static final ValueLayout.OfDouble C_DOUBLE = ValueLayout.JAVA_DOUBLE;
 
         /**
          * The {@code T*} native type.
          */
-        public static final ValueLayout.OfAddress C_POINTER = ValueLayout.ADDRESS.withBitAlignment(64)
-                .withTargetLayout(MemoryLayout.sequenceLayout(C_CHAR));
+        public static final ValueLayout.OfAddress C_POINTER = cPointer(C_CHAR);
 
-        /**
-         * The {@code va_list} native type, as it is passed to a function.
-         */
-        public static final ValueLayout.OfAddress C_VA_LIST = RISCV64.C_POINTER;
     }
+
+    private static ValueLayout.OfAddress cPointer(ValueLayout targetLayoutElementLayout) {
```

Comment:
Maybe there is a better name? `elementLayout` perhaps?

Reply:
I think you can just omit the parameter here - and use JAVA_BYTE. We're not interested in what the layout is - the main thing is that it's unbounded (and that it has minimum alignment). Even using a padding layout of Long.MAX_VALUE would work :-)